src/services/collection.service.js

```diff
@@ -10,19 +12,47 @@ async function getCollectionByIdAndUserId(id, userId) {
   return user;
 }
 
+async function getProduct(pid) {
+  const products = await Products.findOne({
+    where: { id: pid },
+  });
+  if (!products) {
+    return [];
+  }
+  const images = await products.getProductImages();
+  const img = [];
+  for (let i = 0; i < images.length; i += 1) {
+    img.push(images[i].url);
+  }
+  const output = {
+    images: img,
+    product: products,
+  };
+  return output;
+}
+
 async function createCollection(collectionObj) {
   const collection = Collection.create(collectionObj);
   return collection;
 }
 
 async function deleteCollection(collectionId) {
-  const collection = Collection.destroy({ where: { id: collectionId } });
+  await Products.destroy({ where: { collectionId } });
+  const collection = await Collection.destroy({ where: { id: collectionId } });
   return collection;
 }
 
+async function deleteFromCollection(collectionId, productId) {
+  Products.destroy({ where: { id: productId, collectionId } }).then(() => {
+    Images.destroy({ where: { productId } }).then(() => true);
+  });
```

Comment:
once you delete the product, all associated images are deleted
no need to  delete images again

Reply:
check this comment